crates/astria-bridge-withdrawer/src/bridge_withdrawer/ethereum/watcher.rs

```diff
@@ -263,155 +258,242 @@ impl Watcher {
     }
 }
 
-async fn watch_for_sequencer_withdrawal_events(
-    contract: IAstriaWithdrawer<Provider<Ws>>,
-    event_tx: mpsc::Sender<(WithdrawalEvent, LogMeta)>,
-    from_block: u64,
+async fn sync_from_next_rollup_block_height(
+    provider: Arc<Provider<Ws>>,
```

Comment:
unfortunately it needs to be in an `Arc` due to the loop here:
```
    loop {
        select! {
            () = shutdown_token.cancelled() => {
                info!("block watcher shutting down");
                return Ok(());
            }
            block = block_rx.next() => {
                if let Some(block) = block {
                    get_and_send_events_at_block(
                        provider.clone(),
                        contract_address,
                        block,
                        &converter,
                        &submitter_handle,
                    )
                    .await
                    .wrap_err("failed to get and send events at block")?;
                }
            }
        }
    }
```

have to clone it otherwise it gets moved in the previous loop iteration

Reply:
You should be able to change `get_and_send_events_at_block` to take `provider: &Provider<Ws>` too though (and all the other places taking an `Arc`)?